google/resource_compute_target_ssl_proxy.go

```diff
@@ -0,0 +1,247 @@
+			ProxyHeader: proxy_header,
+		}
+		op, err := config.clientCompute.TargetSslProxies.SetProxyHeader(
+			project, d.Id(), proxy_header_payload).Do()
+		if err != nil {
+			return fmt.Errorf("Error updating proxy_header: %s", err)
+		}
+
+		err = computeOperationWait(config, op, project, "Updating Target SSL Proxy")
+		if err != nil {
+			return err
+		}
+
+		d.SetPartial("proxy_header")
+	}
+
+	if d.HasChange("backend_service") {
+		op, err := config.clientCompute.TargetSslProxies.SetBackendService(project, d.Id(), &compute.TargetSslProxiesSetBackendServiceRequest{
+			Service: d.Get("backend_service").(string),
+		}).Do()
+
+		if err != nil {
+			return fmt.Errorf("Error updating backend_service: %s", err)
+		}
+
+		err = computeOperationWait(config, op, project, "Updating Target SSL Proxy")
+		if err != nil {
+			return err
+		}
+
+		d.SetPartial("backend_service")
+	}
+
+	if d.HasChange("ssl_certificates") {
+		sslCertificates, err := expandSslCertificates(d.Get("ssl_certificates").([]interface{}), d, config)
+		if err != nil {
+			return err
+		}
+
+		op, err := config.clientCompute.TargetSslProxies.SetSslCertificates(project, d.Id(), &compute.TargetSslProxiesSetSslCertificatesRequest{
+			SslCertificates: sslCertificates,
+		}).Do()
+
+		if err != nil {
+			return fmt.Errorf("Error updating backend_service: %s", err)
+		}
+
+		err = computeOperationWait(config, op, project, "Updating Target SSL Proxy")
+		if err != nil {
+			return err
+		}
+
+		d.SetPartial("ssl_certificates")
+	}
+
+	d.Partial(false)
+
+	return resourceComputeTargetSslProxyRead(d, meta)
+}
+
+func resourceComputeTargetSslProxyRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	proxy, err := config.clientCompute.TargetSslProxies.Get(
+		project, d.Id()).Do()
+	if err != nil {
+		return handleNotFoundError(err, d, fmt.Sprintf("Target SSL Proxy %q", d.Get("name").(string)))
+	}
+
+	d.Set("name", proxy.Name)
+	d.Set("description", proxy.Description)
+	d.Set("proxy_header", proxy.ProxyHeader)
+	d.Set("backend_service", proxy.Service)
+	d.Set("ssl_certificates", proxy.SslCertificates)
+	d.Set("self_link", proxy.SelfLink)
+	d.Set("proxy_id", strconv.FormatUint(proxy.Id, 10))
+
+	return nil
+}
+
+func resourceComputeTargetSslProxyDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	op, err := config.clientCompute.TargetSslProxies.Delete(
+		project, d.Id()).Do()
+	if err != nil {
+		return fmt.Errorf("Error deleting TargetSslProxy: %s", err)
+	}
+
+	err = computeOperationWait(config, op, project, "Deleting Target SSL Proxy")
+	if err != nil {
+		return err
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func expandSslCertificates(configured []interface{}, d *schema.ResourceData, config *Config) ([]string, error) {
```

Comment:
This function would be a bit simpler if it didn't take the `configured` parameter (which is of a very confusing type). Instead you can use the `d` parameter to extract out the `ssl_certificates` field